File: Leetcode/19. Remove Nth Node From End of List/solution.cpp
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        if (head->next == nullptr) {
            return nullptr;
        }
        
        ListNode* head_p = head;
        ListNode* head_p2 = head;
        
        for (int i = 0; i < n; ++i) {
            head_p = head_p->next;
        }
        
        if (head_p == nullptr) {
            return head->next;
        }
        
        while (head_p->next != nullptr) {
            head_p = head_p->next;
            head_p2 = head_p2->next;
        }
        
        head_p2->next = head_p2->next->next;
        
        return head;
    }
};

File: Leetcode/19. Remove Nth Node From End of List/solution.py
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        if head.next is None:
            return None
        head_p = head
        head_p2 = head
        for _ in range(n):
            head_p = head_p.next
        if not head_p:
            return head.next
        while head_p.next:
            head_p = head_p.next
            head_p2 = head_p2.next
        head_p2.next = head_p2.next.next
        return head
